netCDF4/_netCDF4.pyx

```diff
@@ -1712,7 +1721,7 @@ references to the parent Dataset or Group.
     the parent Dataset or Group.""" 
 
     def __init__(self, filename, mode='r', clobber=True, format='NETCDF4',
```

Comment:
Can we make `filename` optional if `memory` is provided?

Even if netCDF-C needs a bogus `filename` argument, I imagine we could generate something at random.